v0/destinations/webhook/transform.js

```diff
@@ -38,22 +40,42 @@ function process(event) {
     const { headers } = destination.Config;
 
     if (url) {
-      if (method === defaultGetRequestConfig.requestMethod) {
-        response.method = defaultGetRequestConfig.requestMethod;
-        // send properties as query params for GET
-        response.params = getPropertyParams(message);
-      } else {
-        if (method === defaultPutRequestConfig.requestMethod) {
+      switch (method) {
+        case defaultGetRequestConfig.requestMethod: {
+          response.method = defaultGetRequestConfig.requestMethod;
+          response.params = getPropertyParams(message);
+          break;
+        }
+        case defaultPostRequestConfig.requestMethod: {
+          response.method = defaultPostRequestConfig.requestMethod;
+          response.body.JSON = message;
+          response.headers = {
+            "content-type": "application/json"
+          };
+          break;
+        }
+        case defaultPutRequestConfig.requestMethod: {
           response.method = defaultPutRequestConfig.requestMethod;
-        } else {
+          response.body.JSON = message;
+          response.headers = {
+            "content-type": "application/json"
+          };
+          break;
+        }
+        case defaultDeleteRequestConfig.requestMethod: {
+          response.method = defaultDeleteRequestConfig.requestMethod;
+          response.params = getPropertyParams(message);
+          break;
+        }
+        default: {
```

Comment:
We should combine this with the `post` case.

Reply:
Did this for better readability but yeah this adds redundancy

Reply:
This addressed